aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,65 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  // init to column major format
+  // Note: `sol` and `lu` need to be contiguous since we pass
+  // sol.data() and lu.data() to Lapack
+  if (&self == &sol) {
```

Comment:
As we discussed, this is not the comprehensive check. But this is fine as an easy optimization. Can you add a small comment here to explain?

Also, the code still works correctly if you remove this optimization, right?